talisker/metrics.py

```diff
@@ -132,3 +139,175 @@ def inc(self, amount=1, **labels):
             client = talisker.statsd.get_client()
             name = self.get_statsd_name(labels)
             client.incr(name, amount)
+
+
+histogram_archive = 'histogram_archive.db'
+counter_archive = 'counter_archive.db'
+
+
+def histogram_sorter(sample):
+    # sort histogram samples in order of bucket size
+    name, labels, _ = sample
+    return name, float(labels.get('le', 0))
+
+
+def prometheus_cleanup_worker(pid):
+    """Clean up after a multiprocess worker has died."""
+    if prometheus_client is None:
+        return
+
+    try:
+        # just deletes delete gauge files
+        multiprocess.mark_process_dead(pid)
+        prom_dir = os.environ['prometheus_multiproc_dir']
+        worker_files = [
+            'histogram_{}.db'.format(pid),
+            'counter_{}.db'.format(pid),
+        ]
+        paths = [os.path.join(prom_dir, f) for f in worker_files]
+
+        # check at least one worker file exists
+        if not any(os.path.exists(path) for path in paths):
+            return
```

Comment:
how about putting the `os.path.exists()` check in the list-comp for `paths`?

That way we can just check `if not paths` here, and drop the check on `unlink` too?

Reply:
Well, then I have to join twice.
If could do it in two steps, though, will try that